src/other/utils.jl

```diff
@@ -83,3 +83,41 @@ else
 end
 
 funname(c::ComposedFunction) = Symbol(funname(c.outer), :_, funname(c.inner))
+
+"""
+    tforeach(f, x::AbstractArray; basesize::Integer)
+
+Apply function `f` to each entry in `x` in parallel, spawning
+one separate task for each block of at least `basesize` entries.
+
+A number of task higher than `Threads.nthreads()` may be spawned,
+since that can allow for a more efficient load balancing in case
+some threads are busy (nested parallelism).
+"""
+function tforeach(f, x::AbstractArray; basesize::Integer)
+    @static if VERSION >= v"1.4"
+        nt = Threads.nthreads()
+        len = length(x)
+        if nt > 1 && len > basesize
+            # Round size up to ensure all chunks have at least `basesize` entries
+            # This ensures balanced sizes by avoiding a small last chunk
+            basesize′ = cld(len, fld(len, basesize))
+            @sync for p in Iterators.partition(x, basesize′)
```

Comment:
It'd be nice to have a good sense for "common column types" when doing operations like this. For example, any `<:AbstractArray` ends up using `view` for each partition, but for generic iterators, it actually accumulates the partitions manually, which is less efficient.

All that to say, I think we're fine here, it's just something that I pause on because I don't always have a good mental model of what the most common kinds of column types are and whether there might be hidden bottlenecks in things like ths.

Reply:
But columns are always `AbstractArray`?

Reply:
Yes, that's why I think this is really a non-issue; but just gave me a pause to consider if we're really efficient here.

Reply:
OK. BTW, we don't use views are all, we just iterate over relevant indices.